Actor.cs

```diff
@@ -0,0 +1,85 @@
+internal class Actor : MonoBehaviour
+{
+	private static List<Actor> _actors = new List<Actor>();
+	public int StartingHealth;
+	int _currentHealth;
+	private bool _isDead;
+	private CancellationTokenSource _healthCancellationTokenSource;
+	private CancellationTokenSource _deathCancellationTokenSource;
+
+	public event Action<Actor> ActorDied;
+	public event Action<Actor> ActorHealthChanged;
+
+	public void TakeDamage(int damage)
+	{
+		_currentHealth -= damage;
```

Comment:
Probably want a `if(_isDead) return;` statement at the top
Otherwise we'll just repeatedly trigger health changes & death